### Update: pre-existing 1.12.7 SDK test discovery bug + fix

The `openmetadata-service-unit-tests (postgresql)` failure on the first CI run is **not introduced by this PR** — it's a pre-existing bug on the `1.12.7` branch that was latent because the bare-branch CI never runs the unit-test workflow (it only fires on `pull_request` events).

#### Reproduction on `origin/1.12.7` HEAD (no PR changes)

```
$ git checkout origin/1.12.7
$ mvn -pl openmetadata-sdk -am test
[ERROR] Failed to execute goal ... maven-surefire-plugin:3.1.2:test on project openmetadata-sdk:
  org.junit.platform.commons.JUnitException: TestEngine with ID 'junit-jupiter' failed to discover tests:
  ClassSelector [className = 'org.openmetadata.sdk.OMStorageServicesInitTest', ...]
  resolution failed: 'boolean org.junit.platform.commons.util.ReflectionUtils.returnsVoid(java.lang.reflect.Method)'
```

#### Root cause

`openmetadata-sdk/pom.xml` on 1.12.7 hard-pins:

```xml
<junit.version>5.9.3</junit.version>
<junit-platform.version>1.9.3</junit-platform.version>
```

…but the parent pom imports `org.junit:junit-bom:5.11.4` in `dependencyManagement`, which overrides transitive deps to `junit-platform-commons:1.11.4`. `junit-platform-commons` 1.11 made `ReflectionUtils.returnsVoid(Method)` non-public, while `junit-jupiter-engine:5.9.3` still calls it publicly → `JUnitException` during test discovery.

`mvn dependency:tree` confirms the skew on both PR and base:

```
+- org.junit.jupiter:junit-jupiter:jar:5.9.3:test               ← direct, hard-pinned
|  +- org.junit.jupiter:junit-jupiter-api:jar:5.11.4:test       ← transitive, BOM-overridden
|  |  \- org.junit.platform:junit-platform-commons:jar:1.11.4   ← transitive, BOM-overridden
\- org.junit.jupiter:junit-jupiter-engine:jar:5.9.3:test        ← direct, hard-pinned
   \- org.junit.platform:junit-platform-engine:jar:1.11.4       ← transitive, BOM-overridden
```

#### Fix (commit `fb91bbcf`)

Match what `main` already does — derive `junit.version` from the parent's `${org.junit.jupiter.version}` so direct + transitive deps stay aligned:

```diff
- <junit.version>5.9.3</junit.version>
- <junit-platform.version>1.9.3</junit-platform.version>
+ <junit.version>${org.junit.jupiter.version}</junit.version>
+ <junit-platform.version>1.11.4</junit-platform.version>
```

Verified locally:

```
[INFO] Tests run: 195, Failures: 0, Errors: 0, Skipped: 0
[INFO] BUILD SUCCESS
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/openmetadata-clients/openmetadata-java-client/pom.xml
+++ b/openmetadata-clients/openmetadata-java-client/pom.xml
@@ -240,6 +240,7 @@
                             </output>
                             <skipOperationExample>true</skipOperationExample>
                             <skipValidateSpec>true</skipValidateSpec>
+                            <templateDirectory>${project.basedir}/src/main/openapi-templates</templateDirectory>
                         </configuration>
                     </execution>
                 </executions>

--- a/openmetadata-clients/openmetadata-java-client/src/main/openapi-templates/pom.mustache
+++ b/openmetadata-clients/openmetadata-java-client/src/main/openapi-templates/pom.mustache
@@ -1,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>{{groupId}}</groupId>
+    <artifactId>{{artifactId}}</artifactId>
+    <version>{{artifactVersion}}</version>
+    <packaging>jar</packaging>
+</project>

--- a/openmetadata-k8s-operator/pom.xml
+++ b/openmetadata-k8s-operator/pom.xml
@@ -22,8 +22,6 @@
         <java-operator-sdk.version>4.9.2</java-operator-sdk.version>
         <!-- Kubernetes client version (should match openmetadata-service) -->
         <kubernetes-client.version>21.0.1</kubernetes-client.version>
-        <!-- Jackson version (should match parent project) -->
-        <jackson.version>2.17.2</jackson.version>
     </properties>
     
     <dependencies>

--- a/openmetadata-sdk/pom.xml
+++ b/openmetadata-sdk/pom.xml
@@ -17,8 +17,8 @@
     <properties>
         <gson.version>2.10.1</gson.version>
         <okhttp.version>4.12.0</okhttp.version>
-        <junit.version>5.9.3</junit.version>
-        <junit-platform.version>1.9.3</junit-platform.version>
+        <junit.version>${org.junit.jupiter.version}</junit.version>
+        <junit-platform.version>1.11.4</junit-platform.version>
         <mockito.version>5.5.0</mockito.version>
         <slf4j.version>2.0.9</slf4j.version>
         <lombok.version>1.18.30</lombok.version>

--- a/openmetadata-shaded-deps/elasticsearch-dep/pom.xml
+++ b/openmetadata-shaded-deps/elasticsearch-dep/pom.xml
@@ -107,7 +107,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>3.5.1</version>
+                <version>3.6.0</version>
                 <executions>
                     <execution>
                         <phase>package</phase>

--- a/openmetadata-shaded-deps/opensearch-dep/pom.xml
+++ b/openmetadata-shaded-deps/opensearch-dep/pom.xml
@@ -98,7 +98,7 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-shade-plugin</artifactId>
-                <version>3.5.1</version>
+                <version>3.6.0</version>
                 <executions>
                     <execution>
                         <phase>package</phase>

--- a/pom.xml
+++ b/pom.xml
@@ -723,7 +723,7 @@
       <dependency>
         <groupId>org.apache.commons</groupId>
         <artifactId>commons-compress</artifactId>
-        <version>1.26.0</version>
+        <version>1.27.1</version>
       </dependency>
       <dependency>
         <groupId>org.bouncycastle</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -73,7 +73,7 @@
     <mockito.version>5.14.2</mockito.version>
     <!-- Upgrading slf4j causes dropwizard issues -->
     <slf4j.version>2.0.4</slf4j.version>
-    <jackson.version>2.17.2</jackson.version>
+    <jackson.version>2.18.7</jackson.version>
     <dropwizard.version>5.0.0</dropwizard.version>
     <dropwizard-jdbi3.version>5.0.0</dropwizard-jdbi3.version>
     <diffMatch.version>1.0</diffMatch.version>
@@ -95,7 +95,7 @@
     <commons-cli.version>1.9.0</commons-cli.version>
     <commons-io.version>2.17.0</commons-io.version>
     <redshift-jdbc.version>2.1.0.30</redshift-jdbc.version>
-    <gson.version>2.11.0</gson.version>
+    <gson.version>2.13.1</gson.version>
     <mysql.connector.version>9.3.0</mysql.connector.version>
     <postgres.connector.version>42.7.11</postgres.connector.version>
     <jsonschema2pojo.version>1.3.1</jsonschema2pojo.version>
@@ -149,7 +149,7 @@
     <jsonpath.version>2.9.0</jsonpath.version>
     <everit.version>1.14.4</everit.version>
     <json-patch.version>1.13</json-patch.version>
-    <jetty.version>12.1.6</jetty.version>
+    <jetty.version>12.1.7</jetty.version>
     <jakarta-el.version>5.0.0-M1</jakarta-el.version>
     <mcp-sdk.version>1.1.1</mcp-sdk.version>
     <lettuce.version>6.7.1.RELEASE</lettuce.version>
@@ -678,7 +678,7 @@
       <dependency>
         <groupId>com.fasterxml.jackson.dataformat</groupId>
         <artifactId>jackson-dataformat-yaml</artifactId>
-        <version>2.17.2</version>
+        <version>${jackson.version}</version>
         <exclusions>
           <exclusion>
             <groupId>org.yaml</groupId>
@@ -714,6 +714,31 @@
         <groupId>org.flywaydb</groupId>
         <artifactId>flyway-mysql</artifactId>
         <version>${flyway.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.httpcomponents.core5</groupId>
+        <artifactId>httpcore5-h2</artifactId>
+        <version>5.3.5</version>
+      </dependency>
+      <dependency>
+        <groupId>org.apache.commons</groupId>
+        <artifactId>commons-compress</artifactId>
+        <version>1.26.0</version>
+      </dependency>
+      <dependency>
+        <groupId>org.bouncycastle</groupId>
+        <artifactId>bcpkix-jdk18on</artifactId>
+        <version>1.84</version>
+      </dependency>
+      <dependency>
+        <groupId>org.bouncycastle</groupId>
+        <artifactId>bcprov-jdk18on</artifactId>
+        <version>1.84</version>
+      </dependency>
+      <dependency>
+        <groupId>org.bouncycastle</groupId>
+        <artifactId>bcutil-jdk18on</artifactId>
+        <version>1.84</version>
       </dependency>
     </dependencies>
   </dependencyManagement>

